### PR TITLE
fix: support calamari parachain fee calculations

### PR DESCRIPTION
**Pull request by grenade**

### CODE REVIEW

**Comment by TarikGul** [comment]
Hey @grenade thanks for the PR. I'll have a run at this soon, just to verify, but from a quick glance it looks good. Youll just have to run `build.sh` file inside of `/<ROOT>/calc`

**Comment by grenade** [comment]
> Hey @grenade thanks for the PR. I'll have a run at this soon, just to verify, but from a quick glance it looks good. Youll just have to run `build.sh` file inside of `/<ROOT>/calc`

thank you! i've actually not been able to prove this works yet (hence the `draft` pr).
https://api.serrano.calamari.systems/blocks/311056 is running the version of code in this pr but still shows "Fee calculation not supported for this network" so i'm still trying to work out why the fee calculation doesn't appear to work.

**Comment by TarikGul** [comment]
Is there a public node I can test this against?

**Comment by grenade** [comment]
> Is there a public node I can test this against?

yes. any of:
- wss://chilli.calamari.systems
- wss://falafel.calamari.systems
- wss://fritti.calamari.systems
- wss://smoothie.calamari.systems
- wss://jalapeno.calamari.systems

**Comment by TarikGul** [comment]
Could you provide me with a block number/hash that contains an extrinisic that would require a fee as well?

**Comment by TarikGul** [comment]
nvm found a block.

File: src/chains-config/calamariControllers.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+		'RuntimeMetadata',
+		'RuntimeSpec',
+		'TransactionDryRun',
+		'TransactionFeeEstimate',
+		'TransactionMaterial',
+		'TransactionSubmit',
+	],
+	options: {
+		finalizes: true,
+		minCalcFeeRuntime: null,
```
Comment (TarikGul) [inline src/chains-config/calamariControllers.ts:33]: It's this line here that is causing it to error, set it to 1 assuming your minimum runtime version that supports fees is 1.

Comment (grenade) [reply src/chains-config/calamariControllers.ts]: thank you! that makes good sense.

Comment (grenade) [reply src/chains-config/calamariControllers.ts]: with that changed the error message now includes the runtime version. eg: "Fee calculation not supported for 4#calamari" from [/blocks/311056](https://api.serrano.calamari.systems/blocks/311056)

Comment (TarikGul) [reply src/chains-config/calamariControllers.ts]: Hmm, I was able to get it working locally. To test it locally did you link the calc package?

To do that inside of the package.json you will replace the version of `@substrate/calc` to -> `"file:./calc/pkg"` in order to get the updated wasm generated code.

Comment (grenade) [reply src/chains-config/calamariControllers.ts]: thank you again! i really appreciate your help with making this work. that does indeed fix it.

File: calc/pkg/package.json [hunk 2]
```diff
@@ -18,4 +18,4 @@
   "main": "calc.js",
   "homepage": "https://github.com/paritytech/substrate-api-sidecar/tree/master/calc#readme",
   "types": "calc.d.ts"
-}
+}
```
Comment (TarikGul) [inline calc/pkg/package.json:21]: One last nit could you add the space here. When the wasm compiles each time it trims this whitespace.. Sorry abou t that thanks :)

**Review by TarikGul — APPROVED** [review]
lgtm

File: calc/pkg/package.json (hunk 2 above)
Comment (grenade) [reply calc/pkg/package.json]: yes, thanks!

**Review by insipx — APPROVED** [review]

**Comment by TarikGul** [comment]
After the tests, I will merge. Just as a heads up these changes will be reflected in next weeks release.

**Pull request merged**